extensions/sql-database-projects/src/dialogs/publishOptionsDialog.ts

```diff
@@ -77,16 +77,16 @@ export class PublishOptionsDialog {
 				const row = this.optionsTable?.selectedRows![0];
 				const label = this.optionsModel.optionsLabels[row!];
 				await this.descriptionText?.updateProperties({
-					value: this.optionsModel.getDescription(label)
+					value: this.optionsModel.getOptionDescription(label)
 				});
 			}));
 
 			// Update deploy options value on checkbox onchange
 			this.disposableListeners.push(this.optionsTable.onCellAction!((rowState) => {
 				const checkboxState = <azdataType.ICheckboxCellActionEventArgs>rowState;
 				if (checkboxState && checkboxState.row !== undefined) {
-					const label = this.optionsModel.optionsLabels[checkboxState.row];
-					this.optionsModel.optionsLookup[label] = checkboxState.checked;
+					let label = this.optionsModel.optionsLabels[checkboxState.row];
```

Comment:
Have you tried removing all this lookup stuff? You should be able to get the label from the row directly from the data property on the table which you can then use to directly find and update the corresponding option in the model (and at that point you should consider just getting rid of the "model" completely)

Reply:
Here the checkbox onclick action returns below properties and i'm using the azdataType.TableComponent.

checked: false
column: 0
name: 'Include'
row: 10

Reply:
Yes, and then you use the row index to look it up from the table

`const row = this.optionsTable.data[row]`

and pull out the label from whatever column it was set in for that.

Reply:
Got It now what you are referring to was thinking in a loop "the should the table returns the options name!" 

working on eliminating possible local variables, will add it in new iteration.

FYI- I was able to delete the local var booleanOptionMap completely :)